spec.md

```diff
@@ -191,12 +191,93 @@ print("Hello, world!")
+REUSE.toml file. The intended use case of this method is large directories where
+including a comment header in each file (or in `.license` companion files) is
+impossible or undesirable.
+
+A REUSE.toml file MAY be located in any directory, and can cover files that are
+within its directory or deeper. You MAY have multiple REUSE.toml files in
+different directories.
+
+The `version` key (REQUIRED) MUST have an integer value representing the schema
+version of the file. This specification describes version 1 of REUSE.toml.
+
+Each `[[annotations]]` table represents an association of Copyright and
+Licensing Information to zero or more Covered Files. It has the following keys:
+
+- `path` (REQUIRED), a string or list of strings of paths relative to the
+  REUSE.toml file's directory. Each path is a path to a Covered File with which
+  to associate the table's Copyright and Licensing Information. A path MAY fail
+  to resolve to a file in the Project. A path MAY use globbing. These are the
+  globbing and matching rules:
+
+  - `*` matches everything except slashes, including `.` prefixes.
+
+  - `**` matches everything including slashes.
+
+  - `?` matches any single character.
+
+  - `[seq]` matches any character in the sequence.
+
+  - `[!seq]` matches any character not in the sequence.
+
+  - `\` is an escape character.
+
+  - If the path does not contain directory separators (`/`), then any file in
+    the REUSE.toml file's directory or subdirectories that matches the glob or
+    file name is matched.
+
+- `precedence` (OPTIONAL), a literal string. It determines the order of
+  precedence for Copyright and Licensing Information between the REUSE.toml file
+  and Covered Files in the table, and between multiple REUSE.toml files if they
+  both contain Information for the same Covered File. Available values are:
+
+  - `closest`, the default value when `precedence` is not defined. This is an
+    instruction to use the Copyright and Licensing Information inside of the
+    Covered Files or their `.license` files, if available. If no such
+    Information is found, then the Information inside the closest REUSE.toml
+    that covers the File is associated. This is effectively a fallback.
+
+  - `aggregate`. This is an instruction to always associate the Copyright and
+    Licensing Information defined in the table with the table's Covered Files.
+    Subsequently, the `closest` logic is also applied.
+
+  - `toml`. This is an instruction to associate the Copyright and Licensing
+    Information defined in the table with the table's Covered Files, and to
+    ignore any other Information that is closer to the Files. The REUSE.toml
+    that is the closest to the root of the Project is authoritative.
+
+- `SPDX-FileCopyrightText` (REQUIRED), a string or list of strings. Each string
+  MUST be a Copyright Notice to be associated with the table's Covered Files.
+  The prefix of the Copyright Notice MAY be omitted.
```

Comment:
`SPDX-FileCopyrightText` counts as a “prefix” as defined below, so this either allows for the SPDX tag to be part of the string (causing a duplicate tag), or that definition needs to differentiate between the SPDX tag and (other) prefixes.

See my comment on making `SPDX-FileCopyrightText` required ;)

Reply:
I'm not sure what you mean. The goal here is to support the following:

```toml
[[annotations]]
path = "no_prefix.py"
SPDX-FileCopyrightText = "2019 Jane Doe"

[[annotations]]
path = "with_prefix.py"
SPDX-FileCopyrightText = "SPDX-FileCopyrightText: 2019 Jane Doe",
```

The prefixed notices are technically more _correct_, in a sense, but people are going to write without prefixes anyway, because obviously the prefix is already there in the TOML key. We could attempt to force people to add a prefix, but it's hard to succinctly explain, and those kinds of snags don't help adoption.

For additional context:

What's in the TOML key doesn't end up in the REUSE internals, and subsequently not in the SPDX output.

This will all be heaps easier when I get around to https://github.com/fsfe/reuse-tool/issues/947. We could retroactively add the prefix in the REUSE internals.